localstack/services/events/provider.py

```diff
@@ -595,10 +593,33 @@ def filter_event_with_target_input_path(target: Dict, event: Dict) -> Dict:
     return event
 
 
+def process_event_with_input_transformer(target: Dict, event: Dict) -> Dict:
+    """
+    Process the event with the input transformer of the target event,
+    by replacing the message with the populated InputTemplate.
+    docs.aws.amazon.com/eventbridge/latest/userguide/eb-transform-target-input.html
+    """
+    input_transformer = target.get("InputTransformer")
+    if not input_transformer:
+        return event
+    try:
+        input_paths = input_transformer["InputPathsMap"]
+        input_template = input_transformer["InputTemplate"]
+    except KeyError as e:
+        LOG.error("%s key does not exist in input_transformer.", e)
+        raise e
+    for key, path in input_paths.items():
```

Comment:
sidenote: let's see in a follow-up how we can extract a method to reuse this functionality in Pipes 🚀 

For example in [Pipes](https://docs.aws.amazon.com/eventbridge/latest/pipes-reference/API_CreatePipe.html), the top-level parameter is called `InputTemplate`, which confusingly contains another `InputTemplate` alongside some `*Parameters` ([depending on the target](https://docs.aws.amazon.com/eventbridge/latest/pipes-reference/API_PipeTargetParameters.html#eventbridge-Type-PipeTargetParameters-InputTemplate)).